### PR TITLE
Skip setting vlan when parameter is not present

### PR DESCRIPTION
**Pull request by mlguerrero12**
Fixes https://github.com/k8snetworkplumbingwg/sriov-cni/issues/291

### CODE REVIEW

File: pkg/sriov/sriov_test.go [hunk 1]
```diff
@@ -358,7 +358,6 @@ var _ = Describe("Sriov", func() {
 			fakeLink := &utils.FakeLink{LinkAttrs: netlink.LinkAttrs{Index: 1000, Name: "dummylink"}}
 
 			mocked.On("LinkByName", netconf.Master).Return(fakeLink, nil)
-			mocked.On("LinkSetVfVlanQosProto", fakeLink, netconf.VFID, netconf.OrigVfState.Vlan, netconf.OrigVfState.VlanQoS, sriovtypes.VlanProtoInt[sriovtypes.Proto8021q]).Return(nil)
```
Comment (SchSeba) [inline pkg/sriov/sriov_test.go:361]: can you please check that we have unit tests to cover the vlan options? to validate that the mock function was called when needed and doesn't when we don't need it?

**Review by SchSeba — COMMENTED** [review]
great work! 

please just extend a bit more our unit test to cover all the cases related to vlans

```
*conf.Vlan,
*conf.VlanQoS,
sriovtypes.VlanProtoInt[*conf.VlanProto]
```

to be sure we call the function when needed (vlan is not empty)

**Comment by mlguerrero12** [comment]
> great work!
> 
> please just extend a bit more our unit test to cover all the cases related to vlans
> 
> ```
> *conf.Vlan,
> *conf.VlanQoS,
> sriovtypes.VlanProtoInt[*conf.VlanProto]
> ```
> 
> to be sure we call the function when needed (vlan is not empty)

Thanks @SchSeba .

There are many uses related to vlan in config_test.go. Regarding the calling of the function, yes, there is one missing case, so I added unit tests for ApplyVFConfig which covers all optional parameters.

**Comment by mlguerrero12** [comment]
@Eoghan1232, @e0ne, please have a look

**Review by SchSeba — APPROVED** [review]
LGTM

nice work!

**Comment by mlguerrero12** [comment]
Hi @adrianchiris, could you please have a look at this? Thanks!

**Review by Eoghan1232 — APPROVED** [review]

**Comment by Eoghan1232** [comment]
working my way through my emails now, so getting around to the PR's

**Pull request merged**